File: src/main/java/com/chatchatabc/jpademojava/domain/repository/RoleRepository.java
package com.chatchatabc.jpademojava.domain.repository;

import com.chatchatabc.jpademojava.domain.model.Role;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface RoleRepository extends JpaRepository<Role, String> {
    /**
     * Find role by name
     *
     * @param name
     * @return
     */
    Optional<Role> findRoleByName(String name);
}

File: src/main/kotlin/com/chatchatabc/jpademo/domain/repository/RoleRepository.kt
package com.chatchatabc.jpademo.domain.repository

import com.chatchatabc.jpademo.domain.model.Role
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface RoleRepository : JpaRepository<Role, String> {
    /**
     * Find role by name
     */
    fun findRoleByName(name: String): Optional<Role>
}